### PR TITLE
Lt 2507 add locales

### PR DESCRIPTION
**Pull request by gponomarev-lykke**
WIP

### CODE REVIEW

**Comment by gponomarev-lykke** [comment]
I tried to avoid breaking changes as much as possible, so I didn't split into model and entity layers. We'll have to refactor it anyway.

I had to adjust AuditService a little (dropped logs).

The most problematic place is locales upsert logic: 
https://github.com/lykkecloud/chest/pull/72/files#diff-ea910409cde841ecd89357e2daad5f62R29-R95
The overall complexity is high, and there're no safety checks in high load situatuions. It could be refactored to split decision-making and actual calls to the database, but I'd like to introduce tests first.

Also there're some extensions to the Result class that create issues locally. If we agree on those changes and move them to the snow.common, the issues should be resolved.

This PR also introduces ErrorsResponse as a collection of ValidationError.
https://github.com/lykkecloud/chest/pull/72/files#diff-50c8c62b5a25bd084ca4b2dd92f852a3R5
https://github.com/lykkecloud/chest/pull/72/files#diff-244977139a3af02d633db62c03647401R6
It returns {} if a request was successful and if not, it returns this structure

```
{
  "errors": [
    {
      "errorCode": "CannotDeleteDefaultLocale",
      "fieldNames": []
    }
  ]
}
```

```
{
  "errors": [
    {
      "errorCode": "CannotSetLocaleAsDefault",
      "errorMessage": "Must provide localized value for keys: welcome",
      "fieldNames": []
    }
  ]
}
```

File: src/Chest/Data/Repositories/IAuditRepository.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+namespace Chest.Data.Repositories
+{
+    public interface IAuditRepository
+    {
+        Task InsertAsync(IAuditModel model);
+
+        Task<PaginatedResponse<IAuditModel>> GetAll(AuditLogsFilterDto filter, int? skip, int? take);
+    }
+    
+    public class AuditRepository : IAuditRepository
```
Comment (andrey-anokhin) [inline src/Chest/Data/Repositories/IAuditRepository.cs:17]: let's always store interface and implementation in separate files, please

File: src/Chest/Data/Repositories/LocalesRepository.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                await _context.SaveChangesAsync();
+                return new Result<LocalesErrorCodes>();
+            }
+            catch (DbUpdateConcurrencyException e)
+            {
+                if (e.Message.Contains(DoesNotExistException))
+                    return new Result<LocalesErrorCodes>(LocalesErrorCodes.DoesNotExist);
+
+                throw;
+            }
```
Comment (andrey-anokhin) [inline src/Chest/Data/Repositories/LocalesRepository.cs:75]: Dont's we need ClearAllCachedEntries here? And btw we should do smth with that )

File: src/Chest/Services/LocalesService.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+            if (locale.IsDefault)
+            {
+                var keys = await _localizedValuesService.GetMissingKeysAsync(locale);
+                if (keys.Count > 0)
+                    return new ErrorResult<LocalesErrorCodes>(LocalesErrorCodes.CannotSetLocaleAsDefault,
+                        $"Must provide localized value for keys: {string.Join(", ", keys)}");
+
+                // remove default from the old default locale
+                if (defaultLocale != null)
```
Comment (andrey-anokhin) [inline src/Chest/Services/LocalesService.cs:65]: Looks like here we should first check that defaultLocale != null and only after that try to GetMissingKeysAsync

**Review by andrey-anokhin — APPROVED** [review]
Approved, but please fix comments

**Pull request merged**